flow/model/numeric/scale.go

```diff
@@ -13,3 +15,17 @@ func ParseNumericTypmod(typmod int32) (int16, int16) {
 	scale := int16(offsetMod & 0x7FFF)
 	return precision, scale
 }
+
+func DetermineNumericSettingForDWH(precision int16, scale int16, isClickhouse bool) (int16, int16) {
```

Comment:
I think it will be great to have a `WarehouseNumericComaptibility` interface, with implementations for `ClickHouse` , `Snowflake` and `BigQuery`. This interface can have

1. maxPrecision()
2. maxScale()
3. defaultPrecisionAndScale()
4. isValidPrevisionAndScale()

and wire them through.

Reply:
Can this be done as a follow up PR?

Reply:
sure!